### PR TITLE
[WIP] Upgrade to notebook 4.2, ipywidgets 5.0

### PR DESCRIPTION
**Pull request by parente**
Need to wait for ipywidgets 5.0 on conda

### CODE REVIEW

File: pyspark-notebook/Dockerfile [hunk 1]
```diff
@@ -46,19 +46,21 @@ USER jovyan
...
-
+# Activate ipywidgets extension in the environment that runs the notebook server
+RUN jupyter nbextension enable --py widgetsnbextension --sys-prefix
+    
 # Install Python 2 packages and kernel spec
 RUN conda create --quiet --yes -p $CONDA_DIR/envs/python2 python=2.7 \
-    'ipython=4.1*' \
-    'ipywidgets=4.1*' \
+    'ipython=4.2*' \
+    'ipywidgets=5.1*' \
```
Comment (minrk) [inline pyspark-notebook/Dockerfile]: 5.1 or 5.0?

Comment (parente) [inline pyspark-notebook/Dockerfile]: 5.1.2 is already out, hot on the heels of 5.0: https://github.com/ipython/ipywidgets/releases

I figured we'd go to the latest.

Comment (parente) [inline pyspark-notebook/Dockerfile]: And, @jakirkham I'm not up to speed on how these get packaged for conda. Is it now an open community effort with forge or is someone doing official builds of major projects like notebook, ipywidgets, etc.

Comment (minrk) [inline pyspark-notebook/Dockerfile]: Related to that: we should adopt the conda-forge channel in our conda config. #195 should do that.

Comment (willingc) [inline pyspark-notebook/Dockerfile]: @minrk I went ahead and merged #195 since it was an addition. Good question @parente re: builds in conda-forge.

Comment (jakirkham) [inline pyspark-notebook/Dockerfile]: > And, @jakirkham I'm not up to speed on how these get packaged for conda. Is it now an open community effort with forge or is someone doing official builds of major projects like notebook, ipywidgets, etc.

Yeah, there is a lot of discussion on this point. I think the most crucial piece (as we don't all have time to read everything) is this [comment]( https://github.com/conda-forge/conda-forge.github.io/issues/22#issuecomment-197117472 ). There will continue to be a `defaults` channel that is maintained by Continuum. However, I think in practice for projects like Jupyter that release quickly and often, `defaults` won't be the complete solution.

The conda-forge effort is the community driven solution for making and maintaining conda packages. It has been growing quite rapidly. It contains a large portion of what is in [conda-recipes]( https://github.com/conda/conda-recipes ). Also, it is incorporating things where there was no open source recipe previously. I expect we soon will eclipse the `defaults` channel.

The community has also grown quite substantially. Anyone can join and participate as they wish. We have an active working relationship with Continuum and plan to continue that. There are Continuum employees that help with this maintenance burden as well. We have been trying to have regular meetings every two weeks loosely modeled after Jupyter meetings. Details of past and future meetings are in our [hackpad]( https://conda-forge.hackpad.com/conda-forge-meetings-2YkV96cvxPG ).

If there are any more questions, please don't hesitate to ask.

**Comment by jakirkham** [comment]
So, `ipywidgets` 5.1.3 is available from conda-forge now (we skipped a few releases 😉). Going to go ahead an restart this. Are we ok with merging and releasing if it passes?

**Comment by jakirkham** [comment]
CI restarted.

**Comment by jakirkham** [comment]
Seems to be downloading it ok. Looks like we are well on our way. 🚶

**Comment by jakirkham** [comment]
Looks like it passes. Anything else before merging?

**Comment by parente** [comment]
:shipit:

**Comment by willingc** [comment]
> Anything else before merging?

@jakirkham Thank you. :cake:

**Comment by parente** [comment]
@jakirkham I see you already kicked off the build. Thanks!

**Comment by jakirkham** [comment]
Yeah, it is basically done. The push failed the first time so am retrying.

**Comment by jakirkham** [comment]
Alright, it is failing to push one image because it thinks it is already pushing it. I think we need to restart the docker daemon, but I don't know the password for `sudo`.

**Comment by jakirkham** [comment]
Thanks @parente.

**Comment by parente** [comment]
I didn't restart dockerd. It eventually just timed out and the next make release-all ran to completion.

**Comment by jakirkham** [comment]
I see. Weird. Well, it was releasing when I checked in. So, I figured something must have changed. Thanks for taking care of it, @parente.

**Pull request merged**